File: app/src/main/java/com/segyo/memo/ListViewAdapter.java
package com.segyo.memo;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import java.util.ArrayList;

/**
 * Created by sekyo on 2017-12-30.
 */

public class ListViewAdapter extends BaseAdapter {
    private ArrayList<ListViewItem> items = new ArrayList<ListViewItem>() ;

    public void addItem(String content, String time){
        ListViewItem listViewItem = new ListViewItem();
        listViewItem.setContent(content);
        listViewItem.setTime(time);
        items.add(listViewItem);
    }

    @Override
    public int getCount() {
        return items.size();
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        final int pos = position;
        final Context context = parent.getContext();

        if (convertView == null) {
            LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
            convertView = inflater.inflate(R.layout.recycler_view_item_layout, parent, false);
        }

        TextView contentTextView = convertView.findViewById(R.id.content_text_view);
        TextView timeTextView = convertView.findViewById(R.id.time_text_view);

        contentTextView.setText(items.get(position).getContent());
        timeTextView.setText(items.get(position).getTime());

        return convertView;
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public Object getItem(int position) {
        return items.get(position);
    }
}

File: app/src/main/java/com/segyo/memo/ListViewAdapter.kt
package com.segyo.memo

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView
import java.util.ArrayList

/**
 * Created by sekyo on 2018-01-02.
 */
class ListViewAdapter : BaseAdapter() {
    private var items = ArrayList<ListViewItem>()

    fun setItems(items: ArrayList<ListViewItem>) {
        this.items = items
        notifyDataSetChanged()
    }

    override fun getCount(): Int {
        return items.size
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var convertView = convertView
        val pos = position
        val context = parent.context

        if (convertView == null) {
            val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
            convertView = inflater.inflate(R.layout.recycler_view_item_layout, parent, false)
        }

        val contentTextView = convertView!!.findViewById<TextView>(R.id.content_text_view)
        val timeTextView = convertView.findViewById<TextView>(R.id.time_text_view)

        contentTextView.text = items[position].content
        timeTextView.text = items[position].time

        return convertView
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getItem(position: Int): ListViewItem {
        return items[position]
    }
}